src/Umbraco.Web.UI.Client/docs/agentic-workflow.md

```diff
@@ -0,0 +1,457 @@
+# Agentic Workflow
+[← Umbraco Backoffice](../CLAUDE.md) | [← Monorepo Root](../../CLAUDE.md)
+
+---
+
+### Phase 1: Analysis & Planning
+
+**Understand Requirements**:
+1. Read user request carefully
+2. Identify acceptance criteria
+3. Ask clarifying questions if ambiguous
+4. Determine scope (new feature, bug fix, refactor, etc.)
+
+**Research Codebase**:
+1. Find similar implementations in codebase
+2. Identify patterns and conventions used
+3. Locate relevant packages and modules
+4. Review existing tests for similar features
+5. Check for existing utilities or helpers
+
+**Identify Technical Approach**:
+1. Which packages need changes?
+   - New package or modify existing?
+   - Core infrastructure or feature package?
+2. What patterns to use?
+   - Web Component, Controller, Repository, Context?
+   - Extension type? (dashboard, workspace, modal, etc.)
+3. Dependencies needed?
+   - New npm packages?
+   - Internal package dependencies?
+4. API changes needed?
+   - New OpenAPI endpoints?
+   - Changes to existing endpoints?
+
+**Break Down Implementation**:
+1. **Models/Types** - Define TypeScript interfaces and types
+2. **API Client** - Update OpenAPI client if backend changes
+3. **Repository** - Data access layer
+4. **Store/Context** - State management
+5. **Controller** - Business logic
+6. **Element** - UI component
+7. **Manifest** - Extension registration
+8. **Tests** - Unit and integration tests
+9. **Documentation** - JSDoc and examples
+10. **Storybook** (if applicable) - Component stories
+
+**Consider Architecture**:
+- Does this follow project patterns?
+- Are dependencies correct? (libs → packages → apps)
+- Will this create circular dependencies?
+- Is this extensible for future needs?
+- Performance implications?
+
+**Document Plan**:
+- Write brief implementation plan
+- Identify potential issues or blockers
+- Get approval from user if significant changes
+
+### Phase 2: Incremental Implementation
+
+**For New Feature (Component/Package)**:
```

Comment:
The interfaces described below only apply to new extensions, and not necessarily all new features.